### PR TITLE
Initial commit for farmbeats service.

### PR DESCRIPTION
**Pull request by agniveshadhikari**
Adding initial python thin client (convenience layer) code for [Azure Farmbeats](https://github.com/MicrosoftDocs/Azure-FarmBeats/tree/main/documentation).

This is implemented as a thin layer on top of autorest generated client, primarily for:
- serving as the base layer for the data science specific SDK
- having common set of higher level functionality on top of what is provided by the SDK, things like:
    - bulk operations (create, queue jobs)
    - cascade operations on the [farm hierarchy](https://github.com/MicrosoftDocs/Azure-FarmBeats/blob/main/documentation/farm%20hierarchy/readme.md) 
    - perf gains for parallel ops (download of large geospatial files)
- and finally, making it super easy to authenticate to your farmbeats instance (creating the client is a one liner)

WIP. Need to figure out test framework, and any other blockers for this PR.

### CODE REVIEW

**Comment by tjprescott** [comment]
This is the contribution guide for Python: https://github.com/Azure/azure-sdk-for-python/blob/master/CONTRIBUTING.md
I would recommend you start cleaning up the CI failures by fixing your README files.

File: sdk/farmbeats/azure-farmbeats/azure/farmbeats/apigroupclients/boundaryclient.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        boundary_id,
+        boundary
+    ):
+        return self.client.create_boundary(
+            farmer_id,
+            boundary_id,
+            boundary,
+        )
+
+    def get_all(
```
Comment (johanste) [inline sdk/farmbeats/azure-farmbeats/azure/farmbeats/apigroupclients/boundaryclient.py:20]: This methods precludes the ability to support server driven paging. We return `ItemPaged` instances explicitly to not paint ourselves into that corner.

File: sdk/farmbeats/azure-farmbeats-ds/README.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
+# Azure Farnbeats API DS client library for Python
```
Comment (lmazuel) [inline sdk/farmbeats/azure-farmbeats-ds/README.md:1]: This file should not be here

File: sdk/farmbeats/azure-farmbeats/azure/farmbeats/_configuration.py [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+from azure.core.configuration import Configuration
+from azure.core.pipeline import policies
+
+if TYPE_CHECKING:
+    # pylint: disable=unused-import,ungrouped-imports
+    from typing import Any
+
+    from azure.core.credentials import TokenCredential
+
+VERSION = "unknown"
```
Comment (lmazuel) [inline sdk/farmbeats/azure-farmbeats/azure/farmbeats/_configuration.py:18]: Package needs to be generate with `--package-version=1.0.0b1`

File: sdk/farmbeats/azure-farmbeats/azure/farmbeats/_configuration.py [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+        **kwargs  # type: Any
+    ):
+        # type: (...) -> None
+        if credential is None:
+            raise ValueError("Parameter 'credential' must not be None.")
+        super(FarmBeatsClientConfiguration, self).__init__(**kwargs)
+
+        self.credential = credential
+        self.api_version = "2021-03-31-preview"
+        self.credential_scopes = kwargs.pop('credential_scopes', [])
```
Comment (lmazuel) [inline sdk/farmbeats/azure-farmbeats/azure/farmbeats/_configuration.py:42]: This code needs to be generate with `--credential-scope` with your AAD scope

**Comment by ramya-rao-a** [comment]
Closing this PR in favor of https://github.com/Azure/azure-sdk-for-python/pull/18713 which is now merged.
Thanks @iscai-msft 

@agniveshadhikari, Please make further changes in https://github.com/Azure/azure-sdk-for-python/tree/master/sdk/agrifood/azure-agrifood-farming as needed

**Pull request closed without merging**